plugin/core/sessions.py

```diff
@@ -1104,6 +1105,17 @@ def _handle_initialize_error(self, result: Any) -> None:
         # Init callback called after transport is closed to avoid pre-mature GC of Session.
         self.end_async()
 
+    def _get_global_ignore_globs(self, root_path: str) -> List[str]:
+        folder_excludes = [
+            sublime_pattern_to_glob(pattern, is_directory_pattern=True, root_path=root_path)
+            for pattern in globalprefs().get('folder_exclude_patterns')
+        ]
+        file_excludes = [
+            sublime_pattern_to_glob(pattern, is_directory_pattern=False, root_path=root_path)
+            for pattern in globalprefs().get('file_exclude_patterns')
+        ]
+        return folder_excludes + file_excludes + ['**/node_modules/**']
```

Comment:
I guess it's kinda weird that I'm forcing `node_modules` to be ignored but it's kinda a "right" thing to do as nothing should need to watch it and there would be a ton of events triggers on running "npm i" or similar.